### PR TITLE
NDRS-706: Provide JSON versions of Contract and ContractPackage

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-706

Added `Contract` and `ContractPackage` struct to a new `contracts.rs` module inside the json compatibility module.

### CODE REVIEW

**Review by Fraser999 — CHANGES_REQUESTED** [review]
A good start, but a couple of changes required to complete this work.

I'd also like to see an example of a JSON-encoded version of a `Contract` and `ContractPackage` before approving this.

File: node/src/types/json_compatibility/contracts.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// TODO - remove once schemars stops causing warning.
+#![allow(clippy::field_reassign_with_default)]
+
+use datasize::DataSize;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use casper_types::{
+    Contract as DomainContract, ContractHash, ContractPackage as DomainContractPackage,
+    ContractPackageHash, ContractWasmHash, EntryPoint, NamedKey, ProtocolVersion, URef,
```
[the remark this answers is not in the document]
Comment (henrytill) [reply node/src/types/json_compatibility/contracts.rs:10]: I'm fine with @Fraser999's suggestion.  While I had been exploring some options in the space, I ultimately shelved that work in favor of continuing our current approach.

File: types/src/protocol_version.rs [hunk 2]
```diff
@@ -1,15 +1,23 @@
+// TODO - remove once schemars stops causing warning.
```
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply types/src/protocol_version.rs]: Used Version as requested

File: node/src/types/json_compatibility/contracts.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply node/src/types/json_compatibility/contracts.rs]: @Fraser999 I tried wrapping the new types in `key.rs` but the hastle of refactoring was far too much. Multiple tests were breaking and I was unable to get anywhere. I have created an `AccountHash` equivalent directly in the `json_comptability` module itself.

File: node/src/types/json_compatibility/contracts.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    #[data_size(skip)]
+    entry_points: Vec<EntryPoint>,
+    #[data_size(skip)]
+    protocol_version: ProtocolVersion,
+}
+
+impl From<&DomainContract> for Contract {
+    fn from(contract: &DomainContract) -> Self {
+        let entry_points = contract.entry_points().clone().take_entry_points();
+        let named_keys = contract
```
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply node/src/types/json_compatibility/contracts.rs]: Moved into a helper function called `vectorize` in `json_compatibility`

**Comment by darthsiroftardis** [comment]
Closing PR. Moved to: https://github.com/CasperLabs/casper-node/pull/772

**Pull request closed without merging**